bin/varnishd/cache/cache_tcp_pool.c

```diff
@@ -347,27 +350,52 @@ VCP_Open(struct conn_pool *cp, double tmo, const void **privp)
 
 	CHECK_OBJ_NOTNULL(cp, CONN_POOL_MAGIC);
 
+	while (cp->holddown > 0) {
+		Lck_Lock(&cp->mtx);
+		if (cp->holddown == 0) {
```

Comment:
How can it be `== 0` if you need `> 0` to enter the loop?

Should it be this instead? `while (cp->holddown >= 0) {`